blog/source/_posts/Publishing-your-first-boardgame-io-game-in-FBG.md

```diff
@@ -0,0 +1,127 @@
+
+export const checkersGameDef: IGameDef = {
+  code: 'checkers',
+  name: 'Checkers',
+  imageURL: Thumbnail,
+  modes: [{ mode: GameMode.OnlineFriend }, { mode: GameMode.Local }],
+  minPlayers: 2,
+  maxPlayers: 2,
+  description: 'Classic game of Checkers',
+  descriptionTag: `Play Checkers (also known as Draughts) locally 
+  or online against friends!`,
+  instructions: {
+    videoId: 'yFrAN-LFZRU',
+    instructions,
+  },
+  config: () => import('./config'),
+};
+```
+
+Explaining what is going on here:
+
+- _code_: Internal code we are going to use for this game, must match the folder name.
+- _name_: Name that will show up to users.
+- _imageURL_: Tthumbnail used across the app on the home page, game page and lobby. Generally a screenshot of the game, should be 500x250 pixels.
+- _modes_: Each game mode will represent a card in the game page ([check chess here](https://freeboardgame.org/g/chess)).
+  - `GameMode.AI`: Makes the game available offline, allowing users to play single player matches against the computer (more configuration is needed).
+  - `GameMode.OnlineFriend`: Allows users to invite friends to play the game online. Users are going to need internet to connect to the server.
+  - `GameMode.Local`: Makes the game available offline, and players can alternate using the same device to play the game.
+- _minPlayers_ and _maxPlayers_: Count of min and max players allowed for the game.
+- _description_: Short description of the game that will show on the home page card.
+- _descriptionTag_: HTML tag description of the game, specially important for crawlers that can bring organic traffic to your game.
+- _instructions_:
+  - `videoId` is the YouTube video id with a tutorial for the game.
+  - `instructions` allows you import a markdown with written instructions. Also important for crawlers.
+- _config_: This is a function that returns your boardgame.io config. It needs to be a separate file because this will only load all extra resources needed (UI, game logic, etc) if the user wants to play your game.
+
+In `src/games/checkers/media/thumbnail.png`, place:
+![500x250 placeholder](http://www.biotoday.bio/wp-content/uploads/sites/2/2016/01/500x250.png)
+
+In `src/games/checkers/instructions.md`, write:
+
+```markdown
+Instructions for checkers.
+Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+```
+
+Boardgame.io games have two main pieces, its rules (aka `Game`) and the UI (aka `Board`). We need to provide this in the bg.io specific configuration, `src/games/checkers/config.ts`:
+
+```typescript
+import { IGameConfig } from '../index';
+import { CheckersGame } from './game';
+import { Board } from './board';
+
+const config: IGameConfig = {
+  bgioGame: CheckersGame,
+  bgioBoard: Board,
+  debug: true,
+};
+
+export default config;
+```
+
+These two pieces is where the bulk of the game code will live. Let's use placeholders for now so you can see how everything ties together. In `src/games/checkers/board.ts`:
```

Comment:
Shouldn't this be board.tsx?

Reply:
Yes, I haven't finished/tested the article. But thanks for pointing out :) I will probably finish it soon